### PR TITLE
Add a method to convert ErrorDetails instances to dicts

### LINKED ISSUE #79: ErrorDetails instance should be easy to convert to dicts
ErrorDetails instances may be emitted to log streams or returned via APIs, so they should include a convenience method for converting to a simple dict.

### PR DESCRIPTION
**Pull request by chrisrink10**
Fixes #79

### CODE REVIEW

File: src/dataspec/base.py [hunk 1]
```diff
@@ -125,6 +125,38 @@ def with_details(self, tag: Tag, loc: Any = NO_ERROR_PATH) -> "ErrorDetails":
             self.path.insert(0, loc)
         return self
 
+    def as_map(self) -> Mapping[str, Union[str, List[str]]]:
```
Comment (Palisand) [inline src/dataspec/base.py:128]: You may be able to use `TypedDict` here if you want.

File: src/dataspec/base.py [hunk 2]
```diff
@@ -125,6 +125,38 @@ def with_details(self, tag: Tag, loc: Any = NO_ERROR_PATH) -> "ErrorDetails":
...
+
+        ``message`` will remain a string. ``value`` will be passed to :py:func:`str`
+        directly. ``via`` and ``path`` will be returned as a list of strings.
+
+        :return: a mapping of string keys to strings or lists of strings
+        """
+        if isinstance(self.pred, Spec):
+            pred = self.pred.tag
+        elif callable(self.pred):
+            pred = self.pred.__name__
```
Comment (Palisand) [inline src/dataspec/base.py:148]: If `self.pred` is a lambda function, then `pred` will be `"<lambda>"`. I don't think there's much you can do about this except maybe add a comment.

File: src/dataspec/base.py (hunk 1 above)
Comment (chrisrink10) [reply src/dataspec/base.py:128]: That would be nice, but right now we support Python 3.6+ and `TypedDict` is 3.8+.

MyPy raises a bunch of errors if you try to get around it like this:

```python
try:
    from typing import TypedDict
except ImportError:
    ErrorDetailsMap = Mapping[str, Union[str, List[str]]]
else:
    class ErrorDetailsMap(TypedDict):
        ...
```

File: src/dataspec/base.py (hunk 2 above)
Comment (chrisrink10) [reply src/dataspec/base.py:148]: I found this in testing too. Very annoying. I'll add a comment.

**Pull request merged**